sqlglot/optimizer/qualify_columns.py

```diff
@@ -352,6 +375,22 @@ def _add_replace_columns(expression, tables, replace_columns):
         replace_columns[id(table)] = columns
 
 
+def _add_pivot_columns(pivot, source_columns, columns):
+    pivot_output_columns = [col.output_name for col in pivot.args.get("columns", [])]
```

Comment:
this looks expensive, should we move these calls outside the nested loop

Reply:
Ok done